src/autotester/server/utils/tests/file_management_test.py

```diff
@@ -0,0 +1,424 @@
+from autotester.server.utils.file_management import (
+    clean_dir_name,
+    random_tmpfile_name,
+    recursive_iglob,
+    copy_tree,
+    ignore_missing_dir_error,
+    move_tree,
+    copy_test_script_files,
+    fd_open,
+)
+import os.path
+import tempfile
+import shutil
+from autotester.config import config
+import pytest
+from fakeredis import FakeStrictRedis
+from unittest.mock import patch
+
+
+FILES_DIRNAME = config["_workspace_contents", "_files_dir"]
+CURRENT_TEST_SCRIPT_HASH = config["redis", "_current_test_script_hash"]
+
+
+@pytest.fixture
+def empty_dir():
+    """
+    Yields an empty directory
+    """
+    with tempfile.TemporaryDirectory() as empty_directory:
+        yield empty_directory
+
+
+@pytest.fixture
+def dir_has_onefile():
+    """
+    Yields a directory which has only one file
+    """
+    with tempfile.TemporaryDirectory() as dir:
+        with tempfile.NamedTemporaryFile(dir=dir) as file:
+            yield dir, file.name
+
+
+@pytest.fixture
+def dir_has_onedir():
+    """
+    Yields a directory with one sub directory
+    """
+    with tempfile.TemporaryDirectory() as dir:
+        with tempfile.TemporaryDirectory(dir=dir) as sub_dir:
+            yield dir, sub_dir
+
+
+@pytest.fixture
+def nested_fd():
+    """
+    Yields a nested file structure
+    """
+    with tempfile.TemporaryDirectory() as root_dir:
+        with tempfile.TemporaryDirectory(dir=root_dir) as sub_dir1:
+            with tempfile.TemporaryDirectory(dir=sub_dir1) as sub_dir2:
+                with tempfile.NamedTemporaryFile(dir=sub_dir2) as file:
+                    yield root_dir, sub_dir1, sub_dir2, file.name
+
+
+@pytest.fixture(autouse=True)
+def redis():
+    """
+    Mock the redis connection with fake redis and yield the fake redis
+    """
+    fake_redis = FakeStrictRedis()
+    with patch(
+        "autotester.server.utils.redis_management.redis_connection",
+        return_value=fake_redis,
+    ):
+        yield fake_redis
+
+
+@pytest.fixture(autouse=True)
+def empty_tmp_script_dir():
+    """
+    Mock the test_script_directory method and yield an empty temporary directory
+    """
+    with tempfile.TemporaryDirectory() as tmp_dir:
+        files_dir = os.path.join(tmp_dir, "files")
```

Comment:
Do not hard-code the subdirectory name `"files"`. Find where this string is specified and use the variable instead.